backup-stores/azure/src/main/java/io/camunda/zeebe/backup/azure/ManifestManager.java

```diff
@@ -0,0 +1,194 @@
+      }
+      throw new RuntimeException(e);
+    }
+  }
+
+  void completeManifest(final PersistedManifest inProgressManifest) {
+    final byte[] serializedManifest;
+    final var completed = inProgressManifest.manifest().complete();
+    assureContainerCreated();
+    try {
+      serializedManifest = MAPPER.writeValueAsBytes(completed);
+    } catch (final JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+
+    final BlobClient blobClient = blobContainerClient.getBlobClient(manifestPath(completed));
+    try {
+      final Manifest manifest = getManifest(inProgressManifest.manifest().id());
+
+      if (manifest == null) {
+        throw new UnexpectedManifestState("Manifest does not exist.");
+      } else if (manifest.statusCode() != StatusCode.IN_PROGRESS) {
+        throw new UnexpectedManifestState(
+            "Expected manifest to be in progress but was in %s"
+                .formatted(manifest.statusCode().name()));
+      }
+
+      // we only complete the manifest if the eTag matches, thus
+      // assuring no other client has written in the meantime.
+      final BlobRequestConditions blobRequestConditions =
+          new BlobRequestConditions().setIfMatch(inProgressManifest.eTag());
+      blobClient.uploadWithResponse(
+          new BlobParallelUploadOptions(BinaryData.fromBytes(serializedManifest))
+              .setRequestConditions(blobRequestConditions),
+          null,
+          Context.NONE);
+    } catch (final BlobStorageException e) {
+      // will throw precondition failed if etag does not match.
+      if (e.getStatusCode() == PRECONDITION_FAILED) {
+        throw new UnexpectedManifestState(e.getMessage());
+      }
+      throw new RuntimeException(e);
+    }
+  }
+
+  void markAsFailed(final Manifest existingManifest, final String failureReason) {
+    assureContainerCreated();
+    final BlobClient blobClient = blobContainerClient.getBlobClient(manifestPath(existingManifest));
+    final var updatedManifest =
+        switch (existingManifest.statusCode()) {
+          case FAILED -> existingManifest.asFailed();
+          case COMPLETED -> existingManifest.asCompleted().fail(failureReason);
+          case IN_PROGRESS -> existingManifest.asInProgress().fail(failureReason);
+        };
+
+    if (existingManifest != updatedManifest) {
+      try {
+        blobClient.upload(BinaryData.fromBytes(MAPPER.writeValueAsBytes(updatedManifest)), true);
+      } catch (final JsonProcessingException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  Manifest getManifest(final BackupIdentifier id) {
+    assureContainerCreated();
+    final BlobClient blobClient =
+        blobContainerClient.getBlobClient(
+            MANIFEST_PATH_FORMAT.formatted(id.partitionId(), id.checkpointId(), id.nodeId()));
+    if (!blobClient.exists()) {
+      return null;
+    }
+    final BinaryData binaryData = blobClient.downloadContent();
```

Comment:
:upside_down_face: Is there a way to not check for existance first? I'd assume that `downloadContent` would throw an appropriate exception in that case, maybe we can handle that?